### PR TITLE
Container testcases

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

**Comment by Dinaamagdy** [comment]
----------------------------------------------------------------------
Ran 3 tests in 18.384s

OK

File: tests/controller/templates_manager/local_temp/container.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+from jumpscale import j
+from zerorobot.service_collection import ServiceNotFoundError
+from testconfig import config
+import random
+
+class CONTManager:
```
Comment (0xIslamTaha) [inline tests/controller/templates_manager/local_temp/container.py]:
```suggestion
class ContManager:
```

File: tests/controller/templates_manager/local_temp/container.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def __init__(self, parent, service_name=None):
+        self.container_template = 'github.com/threefoldtech/0-templates/container/0.0.1'
+        self._parent = parent
+        self.logger = self._parent.logger
+        self.robot = self._parent.remote_robot
+        self._container_service = None
+        if service_name:
+            try:
+                self._container_service = self.robot.service.get(name=service_name)
+            except ServiceNotFoundError:
```
Comment (0xIslamTaha) [inline tests/controller/templates_manager/local_temp/container.py]: raise the error cause we used the class with `service_name` and we can't find it

File: tests/controller/templates_manager/local_temp/container.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        self._container_service.schedule_action('install').wait(die=wait)
+
+    def uninstall(self, wait=True):
+        self.logger.info('Uninstall {} conatiner'.format(self.container_service_name))
+        self.service.schedule_action('uninstall').wait(die=wait)
+
+    def start(self):
+        return self.service.schedule_action('start')
+
+    def stop(self, force=True):
```
Comment (0xIslamTaha) [inline tests/controller/templates_manager/local_temp/container.py]: You didnt use force arg

File: tests/testcases/a_basics/container_testcases.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+from tests.testcases.base_test import BaseTest
+import unittest
+import time
+from random import randint
+from zerorobot.dsl.ZeroRobotManager import ServiceCreateError
+from tests.controller.controller import Controller
+
+class TESTContainer(BaseTest):
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]:
```suggestion
class TestContainer(BaseTest):
```

File: tests/testcases/a_basics/container_testcases.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: remove this line

File: tests/testcases/a_basics/container_testcases.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    def __init__(self, *args, **kwargs):
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: remove it

File: tests/testcases/a_basics/container_testcases.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        super().__init__(*args, **kwargs)
+
+    def setUp(self):
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: remove it

File: tests/testcases/a_basics/container_testcases.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        super().setUp()
+
+    def test001_create_containers(self):
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: I didnt get the point of creating two containers with the same data

File: tests/testcases/a_basics/container_testcases.py [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+        container_2_data = self.get_container_default_data()
+        self.controller_2.container_manager.install(wait=True, data=container_2_data)
+        self.assertTrue(self.controller_2.container_manager.install_state, " Installtion state is False")
+    
+        self.log('Check that the container have been created.')
+        conts = self.controller.node.client.container.list()
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_1_data['hostname']])
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_2_data['hostname']])
+
+
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: remove it

File: tests/testcases/a_basics/container_testcases.py [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    def test002_create_container_with_all_possible_params(self):
+        """ ZRT-ZOS-002
+        *Test case for creating container with all possible parameters*
+        **Test Scenario:**
+        #. Create a container without providing flist parameter, should fail.
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: This is a separate test case

File: tests/testcases/a_basics/container_testcases.py [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+        container_data["ports"] = ['%s:80'%destination_port]
+        self.controller.container_manager.install(wait=True, data=container_data)
+        self.assertTrue(self.controller.container_manager.install_state, " Installtion state is False")
+
+
+        self.log('Check if the parameters have been reflected correctly')
+        conts = self.controller.node.client.container.list()
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_data['hostname']])
+
+
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: remove

File: tests/testcases/a_basics/container_testcases.py [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+        container_data = self.get_container_default_data()
+        env_name = self.random_string()
+        env_value = self.random_string()
+        destination_port = randint(7000, 8000)      
+        container_data["env"] = [{"name":env_name, "value": env_value}]
+        container_data["ports"] = ['%s:80'%destination_port]
+        self.controller.container_manager.install(wait=True, data=container_data)
+        self.assertTrue(self.controller.container_manager.install_state, " Installtion state is False")
+
+
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: remove

File: tests/testcases/a_basics/container_testcases.py [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+        self.log('Check if the parameters have been reflected correctly')
+        conts = self.controller.node.client.container.list()
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_data['hostname']])
+
+
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_data['hostname']])
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: duplicate

File: tests/testcases/a_basics/container_testcases.py [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+        self.controller_2.container_manager.install(wait=True, data=container_2_data)
+        self.assertTrue(self.controller_2.container_manager.install_state, " Installtion state is False")
+    
+        self.log('Check that the container have been created.')
+        conts = self.controller.node.client.container.list()
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_1_data['hostname']])
+        self.assertTrue([c for c in conts.values() if c['container']['arguments']['hostname'] == container_2_data['hostname']])
+
+
+    def test002_create_container_with_all_possible_params(self):
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: after reviewing this test case, its only check `env` params, It doesnt check `mount`, `storage` .... so please rename it and create another test cases to validate those params

File: tests/testcases/a_basics/container_testcases.py [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+
+    def test001_create_containers(self):
+        """ ZRT-ZOS-001
+        *Test case for creatings container on a zero-os node*
+        **Test Scenario:**
+        #. Create a two container on that node, should succeed.
+        #. Check that the container have been created.
+        """
+        self.logger.info('Create a two container on that node, should succeed.')
+        container_1_data = self.get_container_default_data()
```
Comment (0xIslamTaha) [inline tests/testcases/a_basics/container_testcases.py]: set default data inside the `install` method

**Pull request merged**